kore/src/Kore/Step/ClaimPattern.hs

```diff
@@ -463,3 +465,34 @@ parseRightHandSide term =
       where
         tryPredicate = hush . Predicate.makePredicate
     parsePatternFromTermLike term' = Pattern.fromTermLike term'
+
+-- | The terms of the implication are equivalent in respect to
+-- the associativity of \\and
+areEquivalent
```

Comment:
I think we should do one of two things here: We should either move `areEquivalent` and `toConjunctionOfTerms` to the test suite so that they are never used in the actual code, _or_ we should modify the `Eq` instance for `Pattern` so that it uses this.

We should be _extremely_ careful about using this in tests, because it erases the distinction between predicates and substitutions, which is _very_ important.

Reply:
I think I'll rather move them to the test suite for now. I think we could modify the `Eq` instance for `Pattern`, but I don't know if we ever make decisions based on structural differences in the patterns or not, so it might be better to have a different task for this.